reactor-core/src/test/java/reactor/core/publisher/QueueDrainSubscriberTest.java

```diff
@@ -31,49 +32,41 @@ public class QueueDrainSubscriberTest {
 	@Tag("slow")
 	public void objectPadding() {
 		ClassLayout layout = ClassLayout.parseClass(QueueDrainSubscriber.class);
-		AtomicReference<FieldLayout> wip = new AtomicReference<>();
-		AtomicReference<FieldLayout> requested = new AtomicReference<>();
+
+		AtomicLong currentPaddingSize = new AtomicLong();
+		List<String> fields = new ArrayList<>();
+		List<Long> paddingSizes = new ArrayList<>();
 
 		layout.fields().forEach(f -> {
-			if ("wip".equals(f.name())) wip.set(f);
-			else if ("requested".equals(f.name())) requested.set(f);
+			if (f.name().startsWith("pad")) {
+				currentPaddingSize.addAndGet(f.size());
+			} else {
+				if (currentPaddingSize.get() > 0) {
+					fields.add("[padding]");
+					paddingSizes.add(currentPaddingSize.getAndSet(0));
+				}
+				fields.add(f.name());
+			}
 		});
+		if (currentPaddingSize.get() > 0) {
+			fields.add("[padding]");
+			paddingSizes.add(currentPaddingSize.getAndSet(0));
+		}
 
-		final FieldLayout fieldAfterRequested = layout.fields()
-		                                              .tailSet(requested.get())
-		                                              .stream()
-		                                              .skip(1)
-		                                              .filter(fl -> fl.name().length() >= 4)
-		                                              .findFirst()
-		                                              .get();
-
-		assertThat(layout.fields().headSet(wip.get()))
-				.as("wip pre-padding")
-				.hasSize(15)
-				.allSatisfy(fl -> assertThat(fl.name()).startsWith("p"));
-
-		assertThat(layout.fields().subSet(wip.get(), requested.get()).stream().skip(1))
-				.as("wip-requested padding")
-				.hasSize(15)
-				.allSatisfy(fl -> assertThat(fl.name()).startsWith("p").endsWith("a"));
+		assertThat(fields).containsExactly(
+				"[padding]",
+				"wip",
+				"[padding]",
+				"requested",
+				"[padding]",
+				"cancelled",
+				"done",
+				"actual",
+				"queue",
+				"error"
+		);
 
-		assertThat(layout.fields().subSet(requested.get(), fieldAfterRequested)
-		                 .stream()
-		                 .skip(1))
-				.as("requested post-padding")
-				.hasSize(15)
-				.allSatisfy(fl -> assertThat(fl.name()).startsWith("q").isNotEqualTo("queue"));
-
-		assertThat(wip.get().offset())
-				.as("wip offset")
-				.isEqualTo(136);
-		assertThat(requested.get().offset())
-				.as("requested offset")
-				.isEqualTo(wip.get().offset() + 128);
-
-		System.out.println(wip.get());
-		System.out.println(requested.get());
-		System.out.println(fieldAfterRequested);
+		final long minPaddingSize = 120;
+		assertThat(paddingSizes).filteredOn(padSize -> padSize >= minPaddingSize).hasSize(3);
```

Comment:
filtering is not really useful for the assertion here (and potentially subtly error prone), could be simply `assertThat(paddingSizes).containsExactly(120L, 120L, 120L);`

Reply:
Fixed